### ISSUE TITLE
Proper handling of exceptions

### ISSUE THREAD

**Issue by vbohata** [logstash-plugins/logstash-input-kafka#317]
There are some situations which results to Logstash crash which should be handled by this plugin. They are: temporary timeouts in connections, connection refuses due to group rebalancing, invalid credentials. All these situations caused LS to crash even almost all (except for creds one) are just temporary.

**Comment by cdenneen** [comment]
Yes leaving group due to rebalancing leaves logstash running but kafka plugin not working which results in a "defunct" pipeline and requires manual intervention.

**Comment by colinsurprenant** [comment]
related to elastic/logstash#11603
